src/annotator.ts

```diff
@@ -18,328 +21,444 @@ const includeAsIs = (expr: ast.Expression): boolean => {
+    const rosettaLuaClass = rosetta.luaClasses[cls.name];
+    if (rosettaLuaClass != undefined) {
+        if (func.name === 'new') {
+            // @ts-ignore
+            rosettaObj = rosettaLuaClass.conztructor;
+        } else {
+            rosettaObj = isMethod ? rosettaLuaClass.methods[func.name] : rosettaLuaClass.functions[func.name];
+        }
+    }
+
+    const index = isMethod ? ':' : '.';
+    const name = `${cls.name}${index}${func.name}`;
+
+    out.push('\n');
+
+    if (rosettaObj !== undefined) {
+        let appliedFlags = false;
+
+        if (appliedFlags) {
+            out.push('\n---');
+        }
+
+        const luaParamCount = func.parameters !== undefined ? func.parameters.length : 0;
+        const rosettaParamCount = rosettaObj.parameters !== undefined ? rosettaObj.parameters.length : 0;
+
+        if (
+            func.parameters !== undefined &&
+            rosettaObj.parameters !== undefined &&
+            luaParamCount !== rosettaParamCount
+        ) {
+            throw new Error(
+                "Rosetta's " +
+                    (isMethod ? 'method' : 'function') +
+                    ` '${name}' parameter(s) doesn't match. (lua: ${luaParamCount}, rosetta: ${rosettaParamCount})`
+            );
+        }
+        if (rosettaObj.notes != undefined && rosettaObj.notes.length !== 0) {
+            out.push('\n--- ' + rosettaObj.notes);
+        }
+        if (rosettaParamCount !== 0) {
+            out.push('\n---');
+            for (let index = 0; index < rosettaParamCount; index++) {
+                const param = rosettaObj.parameters[index];
+                let s = `\n--- @param ${param.name} ${param.type.trim()} ${
+                    param.notes != undefined ? param.notes : ''
+                }`;
+                out.push(s);
+            }
+        }
+
+        out.push(`\n---`);
+
+        const returns = (rosettaObj as any).returns;
+        if (returns != undefined) {
+            out.push(`\n--- @return ${returns.type.trim()}${returns.notes != undefined ? ` ${returns.notes}` : ''}`);
+        } else {
+            out.push(`\n--- @return ${returnType}`);
+        }
+
+        if (rosettaObj.parameters != null) {
+            out.push(`\nfunction ${name}(${rosettaObj.parameters.map((o) => o.name).join(', ')}) end`);
+        } else {
+            out.push(`\nfunction ${name}(${func.parameters.join(', ')}) end`);
+        }
+    } else {
+        // Original rendering.
+        out.push(`\n--- @return ${returnType}`);
+        out.push(`\nfunction ${name}(${func.parameters.join(', ')}) end`);
+    }
+};
 
 const annotateFunctionGroup = (cls: LuaClass, functions: LuaFunction[], isMethod: boolean, out: string[]) => {
-    if (functions.length === 0) return
+    if (functions.length === 0) return;
+
+    out.push('\n');
 
-    out.push('\n')
+    let cons;
+
+    functions.sort((a, b) => a.name.localeCompare(b.name));
```

Comment:
Thanks for the alphabetizing here and elsewhere! I'd considered adding this but left it out; I might turn it into an on-by-default option later, but I think it's fine to enforce it, too.